packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -4075,8 +4116,21 @@ function writeBootstrap(
 
 export function writeCompletedRoot(
   destination: Destination,
+  resumableState: ResumableState,
   renderState: RenderState,
 ): boolean {
+  const preamble = renderState.preamble;
+  if (preamble.htmlChunks || preamble.headChunks) {
```

Comment:
The condition here isn't symmetrical with the one in `writePreambleStart`. Maybe that's ok because the ID is essentially reserved and inert without the associated expect link but if we want to actually be precise we need to know if we are in a mode that never emits the paint blocking link

Reply:
You mean the `skipExpect` hack? I override all of writeCompletedRoot in ReactMarkup. That's in line with how the rest of those overrides work.

https://github.com/facebook/react/pull/33016/commits/987bb1fea4b5b20482413511f438d99e0fb44963

However, for the preamble, I still need to write most of it and just exclude this one thing. I'm not sure how to layer that override yet. Maybe it becomes a render state config at some point but for now I wanted it to be compiled out.

Reply:
oh yeah, and the only other place is in the form replaying runtime but that is not part of markup output either?

Reply:
those error if you use functions in forms in markup